### PR TITLE
Removed prometheus-app direct relation in test

### PR DESCRIPTION
**Pull request by natalian98**
Update of [merged prometheus-grafana integration PR](https://github.com/canonical/dex-auth-operator/pull/47) according to latest Observability Team's review: removed the direct relation between prometheus and dex-auth.

### CODE REVIEW

File: tests/integration/test_charm.py [hunk 1]
```diff
@@ -78,14 +83,24 @@ async def test_prometheus_grafana_integration(ops_test: OpsTest):
...
-    response_status = response["status"]
-    log.info(f"Response status is {response_status}")
-    assert response_status == "success"
+    try:
+        for attempt in Retrying(
+            (stop_after_attempt(5) | stop_after_delay(30)),
+            wait=wait_exponential(multiplier=1, min=1, max=10),
+            reraise=True,
+        ):
+            with attempt:
```
Comment (ca-scribner) [inline tests/integration/test_charm.py:92]: Nothing wrong with this approach at all, but to me I found [this way](https://github.com/canonical/minio-operator/pull/56/files#diff-08291d7f4ad7b2caecacb6ae26af540ebbaede9fe42d5e8e1b39d2bdefe4c2d4R112) to make the test a little more readable (it just encapsulates the logic of the retry into a helper so you don't need to look at it unless you're interested).  Your choice whether you use this suggestion or not

Comment (ca-scribner) [reply tests/integration/test_charm.py:92]: To be fair though, the other way has some odd `for: with:` syntax so it isn't perfect either

**Review by ca-scribner — CHANGES_REQUESTED** [review]
A few changes suggested in comments

File: tests/integration/test_charm.py [hunk 2]
```diff
@@ -78,14 +83,24 @@ async def test_prometheus_grafana_integration(ops_test: OpsTest):
...
+                assert response_status == "success"
 
-    response_metric = response["data"]["result"][0]["metric"]
-    assert response_metric["juju_application"] == APP_NAME
-    assert response_metric["juju_model"] == ops_test.model_name
+                response_metric = response["data"]["result"][0]["metric"]
+                assert response_metric["juju_application"] == APP_NAME
+                assert response_metric["juju_model"] == ops_test.model_name
+    except RetryError:
+        log.info("Testing prometheus failed")
```
[the remark this answers is not in the document]
Comment (natalian98) [reply tests/integration/test_charm.py]: That's right, it's corrected now. Thanks!

File: tests/integration/test_charm.py (hunk 1 above)
Comment (natalian98) [reply tests/integration/test_charm.py]: I used your suggestion, that's indeed more readable. Thanks for providing an example

File: src/charm.py [hunk 3]
```diff
@@ -78,7 +78,7 @@ def __init__(self, *args):
 
         self.dashboard_provider = GrafanaDashboardProvider(
             charm=self,
-            relation_name="grafana-dashboards",
+            relation_name="grafana-dashboard",
         )
```
Comment (ca-scribner) [inline src/charm.py:82]: We use the default, so this can be reduced to:
```python
        self.dashboard_provider = GrafanaDashboardProvider(self)
```

**Review by ca-scribner — APPROVED** [review]
lgtm.  You can accept or ignore the one suggestion

**Review by DomFleischmann — APPROVED** [review]

**Pull request merged**